src/config.rs

```diff
@@ -104,16 +104,13 @@ pub(crate) enum ActiveReason {
 
 impl ActiveReason {
     /// Format `ActiveReason` for setting the `RUSTUP_TOOLCHAIN_SOURCE` environment variable.
-    ///
-    /// The format of the variable's content is `source toolchain`. Neither component should
-    /// contain spaces.
-    pub fn to_source(&self, name: &LocalToolchainName) -> String {
+    pub fn to_source(&self) -> &str {
         match self {
-            Self::Default => format!("default {name}"),
-            Self::Environment => format!("env {name}"),
-            Self::CommandLine => format!("cli {name}"),
-            Self::OverrideDB(_) => format!("override {name}"),
-            Self::ToolchainFile(_) => format!("config {name}"),
+            Self::Default => "default",
+            Self::Environment => "env",
+            Self::CommandLine => "cli",
+            Self::OverrideDB(_) => "override",
+            Self::ToolchainFile(_) => "config",
```

Comment:
> This is redundant, because the latter component is RUSTUP_TOOLCHAIN. I'll fix this PR in the next few hours.

Huh, `RUSTUP_TOOLCHAIN` is documented as something the user can set; I didn't realize that rustup sets it on child processes but https://github.com/rust-lang/rustup/blob/061cf7830a61eaefaddddb8454232186d26fd2d1/src/toolchain.rs#L184